src/main/kotlin/com/exactpro/th2/codec/json/JsonPipelineCodec.kt

```
@@ -141,14 +136,14 @@ class JsonPipelineCodec : IPipelineCodec {
     override fun encode(messageGroup: MessageGroup): MessageGroup {
         val messages = messageGroup.messagesList
 
-        if (messages.isEmpty() || messages.none { it.message.metadata.protocol == PROTOCOL }) {
-            return messageGroup
-        }
-
         val builder = MessageGroup.newBuilder()
 
         for (message in messages) {
-            if (message.message.metadata.protocol != PROTOCOL) {
+            if (!message.hasMessage()) {
+                builder.addMessages(message)
+                continue
+            }
```

Maybe we should report an error in this case?